app/lib/katello/validators/kickstart_repository_validator.rb

```diff
@@ -0,0 +1,33 @@
+module Katello
+  module Validators
+    class KickstartRepositoryValidator < ActiveModel::Validator
+      def validate(record)
+        ks_repo_id = kickstart_repository_id(record)
+        return unless ks_repo_id
+
+        msg = if !record.operatingsystem.blank?
+                _("Please select an operating system before assigning a kickstart repository")
+              elsif !record.architecture.blank?
+                _("Please select an architecture before assigning a kickstart repository")
+              elsif !record.operatingsystem.is_a?(Redhat)
+                _("Kickstart repositories can only be associated with Red Hat hosts")
```

Comment:
Might need to work on the messaging. Because Centos will also be considered host belonging to redhat family even though I wouldnt call it a RH host ,.. May be make that distinction (host belonging to RH family.)